Neos.ContentRepository.Core/Classes/Feature/Common/TetheredNodeInternals.php

```diff
@@ -148,4 +160,206 @@ protected function createEventsForMissingTetheredNode(
+                        $creationOrigin->toDimensionSpacePoint(),
+                    )
+                ) {
+                    VariantType::TYPE_SPECIALIZATION => new NodeSpecializationVariantWasCreated(
+                        $contentGraph->getWorkspaceName(),
+                        $contentGraph->getContentStreamId(),
+                        $nodeAggregateId,
+                        $creationOrigin,
+                        $originDimensionSpacePoint,
+                        $interdimensionalSiblings,
+                    ),
+                    VariantType::TYPE_GENERALIZATION => new NodeGeneralizationVariantWasCreated(
+                        $contentGraph->getWorkspaceName(),
+                        $contentGraph->getContentStreamId(),
+                        $nodeAggregateId,
+                        $creationOrigin,
+                        $originDimensionSpacePoint,
+                        $interdimensionalSiblings,
+                    ),
+                    default => new NodePeerVariantWasCreated(
+                        $contentGraph->getWorkspaceName(),
+                        $contentGraph->getContentStreamId(),
+                        $nodeAggregateId,
+                        $creationOrigin,
+                        $originDimensionSpacePoint,
+                        $interdimensionalSiblings,
+                    ),
+                }
+                : new NodeAggregateWithNodeWasCreated(
+                    $contentGraph->getWorkspaceName(),
+                    $contentGraph->getContentStreamId(),
+                    $nodeAggregateId,
+                    $tetheredNodeTypeDefinition->nodeTypeName,
+                    $originDimensionSpacePoint,
+                    $interdimensionalSiblings,
+                    $parentNodeAggregateId,
+                    $tetheredNodeTypeDefinition->name,
+                    $defaultValues,
+                    NodeAggregateClassification::CLASSIFICATION_TETHERED,
+                );
+
+            $creationOrigin ??= $originDimensionSpacePoint;
+        }
+
+        foreach ($tetheredNodeType->tetheredNodeTypeDefinitions as $childTetheredNodeTypeDefinition) {
+            $events = array_merge(
+                $events,
+                iterator_to_array(
+                    $this->createEventsForMissingTetheredNodeAggregate(
+                        $contentGraph,
+                        $childTetheredNodeTypeDefinition,
+                        $affectedOriginDimensionSpacePoints,
+                        $coverageByOrigin,
+                        $nodeAggregateId,
+                        null,
+                        $nodeAggregateIdsByNodePaths,
+                        $currentNodePath->appendPathSegment($childTetheredNodeTypeDefinition->name),
+                    )
+                )
+            );
+        }
+
+        return Events::fromArray($events);
+    }
+
+    protected function createEventsForWronglyTypedNodeAggregate(
```

Comment:
I can't see where this is ever called? It calls itself recursively, but outside of that?

Reply:
It's called here:
https://github.com/neos/neos-development-collection/blob/c8f13d35ac1ffce6eb327c67418459d5dae79ac6/Neos.ContentRepository.Core/Classes/Feature/NodeTypeChange/NodeTypeChange.php#L268-L268